app/src/main/java/com/github/se/polyfit/ui/components/dialog/PictureDialog.kt

```diff
@@ -21,21 +20,18 @@ import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import com.github.se.polyfit.R
 
-// TODO : make the button functions into a List
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun PictureDialog(
     onDismiss: () -> Unit,
-    onFirstButtonClick: () -> Unit,
-    onSecondButtonClick: () -> Unit,
-    firstButtonName: String,
-    secondButtonName: String
+    onButtonsClick: List<() -> Unit>,
+    buttonsName: List<String>,
 ) {
+  assert(buttonsName.size == onButtonsClick.size)
```

Comment:
Having an `assert` is a great idea for testing, but it might not be ideal to have that in the main code. It might be a good idea to either remove it or use the Kotlin function `require`.

Reply:
:+1: